lib/HTML/Restrict.pm

```diff
@@ -276,11 +281,32 @@ sub _build_parser {
         ],
 
     );
-
 }
 
 sub process {
+    my $self = shift;
 
+    my $cleaned = $self->_process(@_);
+    return $cleaned if !$cleaned;
+
+    my $previous_iteration = $cleaned;
+
+    my $i = 1;    # We already cleaned once just above
+
+    while ( $i < $self->max_parser_loops ) {
+        $i++;
+        my $new = $self->_process($previous_iteration);
+        last if $new eq $previous_iteration;
+        if ( $i == $self->max_parser_loops ) {
+            die sprintf( 'Could not clean input after %s attempts', $i );
+        }
+
+        $previous_iteration = $new;
+    }
+    return $self->_processed;
```

Comment:
It would be a bit more efficient (and maybe clearer) to just return `$previous_iteration`. The state in the object is a bit weird.

Reply:
Yeah, the object state is complicated.  I'm now just returning `$previous_iteration`.